### PR TITLE
Feature/lxl 3589 link agents 4

### PR DESCRIPTION
**Pull request by kwahlin**
Last changes now! See spec https://jira.kb.se/browse/LXL-3589

### CODE REVIEW

File: whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy [hunk 1]
```diff
@@ -94,14 +96,20 @@ Map loadAgents() {
                 agents[chip] << id
             else
                 agents[chip] = [id] as Set
+
+            chip = chip.findAll { it.key != '@type' }
```
Comment (lrosenstrom) [inline whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:100]: Maybe do `chip.remove('@type')` for clarity?

File: whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy [hunk 2]
```diff
@@ -94,14 +96,20 @@ Map loadAgents() {
...
+            if (agents.containsKey(chip))
+                agents[chip] << id
+            else
+                agents[chip] = [id] as Set
+
         }
```
Comment (lrosenstrom) [inline whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:106]: Do you really want to store one agent with '@type' and one without?

File: whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy (hunk 1 above)
Comment (kwahlin) [reply whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:100]: Would prefer that too but it'll mutate `chip` so we wouldn't get two distinct map entries (one with `@type` and one without).

File: whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy (hunk 2 above)
Comment (kwahlin) [reply whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:106]: One could probably come up with a more clever solution, I just implemented the first somewhat (time) efficient solution that came to mind :)

Comment (lrosenstrom) [reply whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:106]: Sure, if it works the way you planned it's perfectly OK!

**Review by lrosenstrom — APPROVED** [review]
:ship:

File: whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy (hunk 1 above)
Comment (lrosenstrom) [reply whelktool/scripts/cleanups/2022/01/lxl-3589-link-agents.groovy:100]: Ah, of course :+1:

**Pull request merged**